### PR TITLE
fix: Workspace members search fixes for users and groups [WEB-753]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

Fixes two bugs related to the Workspace Add Member modal:

- Autocomplete was not matching any users or groups now that they're more complex UI objects. In this fix, replaces `handleFilter`'s `any`-type prop with values which we expect for a User or Group item. This makes it easier to follow the function for search term -> items match

Screen Shot 2022-12-22 at 10 57 34 AM

- ~Remove Member dropdown was lowercase "remove"~ fixed by #5640

- Third issue which I don't know how to solve? The members search rotates between an up and down arrow, but also changes to a search icon, so we get an upside-down search icon. On the [antd reference page](https://ant.design/components/select), this would only change to a search icon. So is this a bug with our version of antd, could CSS block the rotation..?

Screen Shot 2022-12-22 at 11 01 11 AM

## Test Plan

- Run determined-ee, with RBAC enabled.
- Log in as admin and create a workspace
- In the workspace's Members tab, click "Add Members"
- In the Add Members modal, click inside the search but don't type any search. You should have a list of all addable groups and users. If you don't have any other users or groups in this instance, create those to continue testing.
- In the Add Members search, type letters which don't match any group or user. You should see "No data" empty results.
- In the Add Members search, type a group name, a username, and a user's display name. Confirm that the filter keeps a matching user/group and hides others.
- Select one member and add them to the workspace
- In the workspace's members tab, remove them from the workspace (menu should say Remove)

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/pages/WorkspaceDetails/WorkspaceMembers.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ const GroupOrMemberActionDropdown: React.FC<GroupOrMemberActionDropdownProps> =
 
   const menuItems: DropDownProps['menu'] = useMemo(() => {
     const MenuKey = {
-      Remove: 'remove',
+      Remove: 'Remove',
```
Comment (keita-determined) [inline webui/react/src/pages/WorkspaceDetails/WorkspaceMembers.tsx:70]: i made the change in[ the different PR](
https://github.com/determined-ai/determined/pull/5640/files#diff-a9a21494cf20b899ce8dc5a2baa98d4b53f28afe1d3ea1ab63d30895b43b14b3R84)

and i noticed this change breaks the code

Comment (mapmeld) [reply webui/react/src/pages/WorkspaceDetails/WorkspaceMembers.tsx:70]: I agree, your PR fixes this in the correct place (label) and my change here doesn't make sense. I'll get a review to you on that PR soon and then work on getting that in first

File: webui/react/src/hooks/useModal/Workspace/useModalWorkspaceAddMember.tsx [hunk 2]
```diff
@@ -59,15 +68,20 @@ const useModalWorkspaceAddMember = ({
 
   const handleFilter = useCallback(
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (gt2345) [inline webui/react/src/hooks/useModal/Workspace/useModalWorkspaceAddMember.tsx:70]: nit: no-explicit-any is no longer needed

**Review by gt2345 — APPROVED** [review]
LGTM, thank you for fixing this

**Pull request merged**